### PR TITLE
Integrate cli tool by spearson78 for handling repos

### LINKED ISSUE #115: Backend for git repositories
We do not use Discussions anymore, but:

### Discussed in https://github.com/NostrGit/NostrGit/discussions/73

<div type='discussions-op-text'>

<sup>Originally posted by **AbhinavMir** March  5, 2023</sup>
I'm assuming the stack would be something like nostr, gitscm (handled by say, Go) hosting git on ipfs or self-hosted servers -> api exposed -> react renders the API.</div>

Let's continue the discussion here :heart:

### PR DESCRIPTION
**Pull request by cypherhoodlum**
### What does it do? Why?

Closes #115.

Integrates the [gitnostr](https://github.com/spearson78/gitnostr) repository to our codebase. Enables creating and cloning repositories with the git-nost-cli tool. The repositories are hosted on nostr relays. 

- Created a Dockerfile for the git-nostr-bridge.
- Created a docker-compose.yml to get git-nostr-bridge running conveniently.
- Added detailed instructions to readme.md

### QA

The following command (in project root)
```bash
$ docker compose up
```
should run a container called `git-nostr-bridge`. 

The user should now be able to use their git-nostr-cli with the following configuration:

```JSON
{
    "relays": ["<relay addresses>"],
    "privateKey": "<user's private key in hex>",
    "gitSshBase": "root@localhost"
}
```

### Review

- All GHA are success
- Use https://ui.shadcn.com/docs often as possible
- If Vercel build fails, run `yarn build` locally to get the same errors
      more to come...

### CODE REVIEW

**Comment by cypherhoodlum** [comment]
After running the containers http://localhost:8080 returns the default next.js 404 error template for some reason. Might be a problem with the nginx.conf file, which is just pointing to / at the moment.

**Comment by cypherhoodlum** [comment]
It seems this could be the reason why all paths returns a next.js 404 error. From vercel/next.js repo:
 
Screenshot from 2023-03-17 21-50-56

This is not an issue when served through the vercel pipeline like we do on nostrgit.com.

**Comment by cypherhoodlum** [comment]
I think it would be good to have just one command `docker compose up` which sets up git-nostr-bridge and the frontend all at once (what I'm trying to accomplish in this PR). What's the plan to make our repo self-hostable as it is now if the app must be deployed through a github-integrated vercel pipeline to be reachable? :thinking:

**Comment by PeerRich** [comment]
just fyi vercel doesnt run docker

**Comment by cypherhoodlum** [comment]
> just fyi vercel doesnt run docker

Well, damn :-1:

**Comment by cypherhoodlum** [comment]
Doesn't matter, we can run just the git-nostr-bridge in a container and forget about putting the frontend into a container. We just need to make sure the relays are the same in git-nostr-bridge configuration and in the frontend so that the frontend can fetch repositories from the correct relays.

File: docker-compose.yml
Comment (b35363) [inline docker-compose.yml:1]: you shouldn't use port 22 for ssh post synchronization, use another port
Overall I find the idea pretty interesting, gitnostr was a great pretender last months.

```yaml
ports:
 - 2202:22
```

File: gitnostr/Dockerfile
Comment (b35363) [inline gitnostr/Dockerfile:1]: Why not using a specific user ?  **root** should never be used, unless it's absolutely necessary !
You can create sudo users or just users with go or any tools needed.

This is not ready for production ;
[] Should use fixed versions, projects may not run on latest until some time
[] Should create and use a non-root user
[] Should use alpine images, more lightweight

I'll come back with the changed Dockerfile, don't worry with this

**Review by b35363 — APPROVED** [review]

File: docker-compose.yml
Comment (cypherhoodlum) [reply docker-compose.yml:1]: > you shouldn't use port 22

Good point.

> pretender

Do you mean contender? :smile:

File: gitnostr/Dockerfile
Comment (cypherhoodlum) [reply gitnostr/Dockerfile:1]: Fantastic, @b35363! :rocket:  I was using root as a quick and very dirty fix to a file permission issue I had when testing this out. This whole PR is meant to be a POC that this integration actually works and that we can use spearson78's solution. A user must be created instead as you say. Thanks for doing it if you find the time!

**Pull request merged**